ecommerce/api.py

```diff
@@ -266,15 +266,25 @@ def get_valid_coupon_versions(
     product_coupon_subquery = CouponEligibility.objects.select_related("coupon").filter(
         product=product, coupon__enabled=True
     )
+
+    # Global should apply only to 100% coupons (amount=1.0)
+    global_coupon_subquery = Coupon.objects.filter(
+        is_global=True, enabled=True, payment__versions__amount=decimal.Decimal(1)
```

Comment:
does it not be under `full_discount` check ?

Reply:
Global coupons should only be applied to 100% off only. That's why this query includes a `payment__versions__amount=decimal.Decimal(1)` filter.